### ISSUE TITLE
Phel getting started templates

### ISSUE THREAD

**Issue by mabasic** [phel-lang/phel-lang#192]
The idea is that instead of doing all the work that is currently on [Getting Started](https://phel-lang.org/documentation/getting-started/) documentation page we would have two templates:

- blank template
- web template

**blank template** - Creates a Phel application which has the structure for creating libraries or command line programs.

**web template** - The same as blank template, but adds `index.php` file which can be served with `php -S localhost:8000 ./src/index.php`

I have created an example repository for "blank template". You can view it here: https://github.com/mabasic/phel-blank-template.

Let me know what you think. If this is something that Phel should have? I am open to moving the repository to phel-lang and doing changes to it.

Basically, people would have to run this one command and they would have Phel application already set up, with all dependencies installed:

```
composer create-project mabasic/phel-basic-template example-app dev-master
```

_Version constraint `dev-master` would not be needed when a stable release is released._

The command could look like: `composer create-project phel-lang/blank example-app`

The Getting Started chapter would have sections:

- Installation (templates: blank, web, docker, ...)
- Manual installation
- REPL
- Editor Support

**Comment by jenshaase** [comment]
@mabasic Sorry I'm not so much into composer project templates. Do you need to tag these templates somehow or can it be any packagist repository.  I mean we already have https://github.com/phel-lang/phel-scaffolding. Can I do this now?
```
composer create-project phel-lang/phel-scaffolding example-app dev-master
```
What are the differences?

Any idea on how to automatically test these templates against the current master version? We should somehow be notified if we change something in master and the template needs to be updated. Otherwise, I may happen that we release a new version but the Getting started guide is not working.

**Comment by mabasic** [comment]
> @mabasic Sorry I'm not so much into composer project templates. Do you need to tag these templates somehow or can it be any packagist repository.  I mean we already have https://github.com/phel-lang/phel-scaffolding. Can I do this now?
> ```
> composer create-project phel-lang/phel-scaffolding example-app dev-master
> ```

You can, but see bellow.

> What are the differences?

`phel-scaffolding` repository includes a whole lot: docker, examples, a bunch of namespaces in composer.json, a build directory, a .github directory.

Where my idea for a great starting point is the the result of the getting started guide. Src folder with boot.phel file, and a tests folder with an example test. That is all. If you were to build a phel library, running that single command would provide yiu with with everything needed and no bloat. 

If you were to complete the getting started manual step by step, you would end up with the same files and structure.

We could add index.php inside, but if your intention was a cli app, index.php would be bloat. For that purpose I would have two templates:

- blank aka cli (without index.php)
- web (with index.php file)

> Any idea on how to automatically test these templates against the current master version? We should somehow be notified if we change something in master and the template needs to be updated. Otherwise, I may happen that we release a new version but the Getting started guide is not working.

Since we would have a single command we could use version constraint to be sure that the command matches the documentation.

For example:

Phel version 0.1.0 would have this command in the getting started documentation:

```
composer create-project phel-lang/cli "0.1.0"
```

Which would match Phel version 0.1.0 and install that phel version.

If Phel 1.0.0 was released with completely different folder structure and requirements, we would change the documentation to use version constraint "1.0.0" and we would be sure that it works.

We could omit version constraint to install the latest version of phel.

So for example, we couod have in the documentation:

composer create-project phel-lang/web

Which would install the latest version of phel. We would have to set the latest phel version in composer.json manually in the master branch. Whatever we set there, that would be installed. 

Since the template can be installed, we could automate it so that if the requirement can be installed and the example test could be run, the template is valid.

**Comment by jenshaase** [comment]
Ok, I see your point. The code in the repository is much less then in `phel-scaffolding`. I will leave this open for discussions. We should evaluate all options and find the best way to get people started with Phel. But first I will release the first version.

**Comment by mabasic** [comment]
I've updated my getting started templates:

https://github.com/mabasic/phel-cli

```
composer create-project mabasic/phel-cli example-app dev-master
cd example-app
./vendor/bin/phel run src/boot.phel
```

https://github.com/mabasic/phel-web

```
composer create-project mabasic/phel-web example-app dev-main
cd example-app
php -S localhost:8000 -t ./public/
```

**Comment by mabasic** [comment]
I've updated my templates to use the latest phel version 0.4
